frontend/src/host_orchestrator/orchestrator/userartifacts.go

```diff
@@ -24,65 +25,86 @@ import (
 	"os/exec"
 	"path/filepath"
 	"strings"
+	"sync"
 
 	apiv1 "github.com/google/android-cuttlefish/frontend/src/host_orchestrator/api/v1"
 	hoexec "github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator/exec"
 	"github.com/google/android-cuttlefish/frontend/src/liboperator/operator"
+	"github.com/google/uuid"
 )
 
 // Resolves the user artifacts full directory.
 type UserArtifactsDirResolver interface {
 	// Given a directory name returns its full path.
-	GetDirPath(name string) string
+	GetDirPath(name string, isLegacy bool) string
 }
 
 type UserArtifactChunk struct {
-	Name             string
-	File             io.Reader
-	UseChunkOffset   bool
-	ChunkOffsetBytes int64
-
-	ChunkNumber    int   // Deprecated: use `ChunkOffsetBytes`
-	ChunkTotal     int   // Deprecated: use `OffsetBytes`
-	ChunkSizeBytes int64 // Deprecated: use `OffsetBytes`
+	Name          string
+	File          io.Reader
+	OffsetBytes   int64
+	SizeBytes     int64
+	FileSizeBytes int64
```

Comment:
Please create two different structs to pass as parameters to `UpdateArtifact` and `UpdateArtifactsWithDir`